### PR TITLE
Application shutdown problem fix

### PR DESCRIPTION
**Pull request by paskal**
- fix a few things in Readme
- shutdown all services before stopping the application (previously shutting down services was a non-blocking operation and likely did not complete properly)
- fix `image.Service.Close()` as previously it was stuck for eternity in case `image.Service.Submit` was not called at least once during the application run

This PR likely will improve the situation with the problem described in #584 as previously `image.Service.Close()` likely was not even called on program shutdown.

### CODE REVIEW

File: backend/app/store/image/image.go [hunk 1]
```diff
@@ -150,13 +151,16 @@ func (s *Service) Cleanup(ctx context.Context) {
...
 	log.Printf("[INFO] close image service ")
-	atomic.StoreInt32(&s.term, 1) // enforce non-delayed commits for all ids left in submitCh
-	for {
-		// set to 0 by Commit goroutine after everything waited on TTL sent
-		if atomic.LoadInt32(&s.term) == 0 {
-			break
+	// terminate commit goroutine using s.term only if it is started
+	if atomic.LoadInt32(&s.term) == 1 {
+		atomic.StoreInt32(&s.term, 0) // enforce non-delayed commits for all ids left in submitCh
+		for {
```
Comment (umputun) [inline backend/app/store/image/image.go:157]: not related yo your changes but a general suggestion - Close is blocking operation which may take a long time. Maybe we want to pass ctx for early termination?

Comment (paskal) [reply backend/app/store/image/image.go:157]: It's an interesting point. I would suggest merging this PR as it makes something which worked fast but unreliable working slower but 100% reliable.

---

Adding context cancelation which later would be just checked by Close() function makes things more complex but would speed them up.

As far as I see one of the concerns is that we need to wait for an HTTP server to stop first to prevent crippled answers to requests in progress, which would leave us with everything but HTTP server to shut down. That means that we can parallelize only things that terminate later in code.
Both with or without concern above I don't know how much time we really save on parallelizing these things: slowest operation is disk IO, so `image.Store.Close()` moving files on HDD from staging to permanent storage would take magnitudes higher amount of time than any other operation, driving effects of parallelization to zero. And un scenario `image.Store.Close()` doesn't need to do anything or moves files fast on SSD, we are saving previous calls unparalleled execution time which is unknown and we shouldn't care about until we really see a problem there.

In summary, I think it's premature optimization and we shouldn't care about it yet.
